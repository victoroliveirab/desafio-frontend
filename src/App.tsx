import { BrowserRouter } from 'react-router-dom';
import AuthProvider from 'shared/providers/auth';
import Pages from 'pages';
import Header from 'components/Header';

function App() {
  return (
    <BrowserRouter>
      <AuthProvider>
        <Header />
        <Pages />
      </AuthProvider>
    </BrowserRouter>
  );
}

export default App;
