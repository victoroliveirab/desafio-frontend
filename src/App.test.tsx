import App from 'App';
import { render } from 'test-utils';

describe('<App />', () => {
  beforeEach(() => {
    const mockIntersectionObserver = jest.fn();
    mockIntersectionObserver.mockReturnValue({
      observe: () => null,
      unobserve: () => null,
      disconnect: () => null,
    });
    window.IntersectionObserver = mockIntersectionObserver;
  });

  test('should match snapshot', () => {
    const component = render(<App />);

    expect(component).toMatchSnapshot();
  });
});
