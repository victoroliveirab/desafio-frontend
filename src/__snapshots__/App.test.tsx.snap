// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="row"
        >
          <div
            class="header-logo"
          >
            <svg>
              logo.svg
            </svg>
            <h1>
              FakeTube
            </h1>
          </div>
          <button
            type="button"
          >
            Login
          </button>
        </div>
        <hr />
        <div
          class="row"
        >
          <div
            class="menu"
          >
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              href="/"
            >
              Home
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              href="/videos"
            >
              Videos
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              href="/channels"
            >
              Channels
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              href="/history"
            >
              History
            </a>
            <a
              class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
              href="/upload"
            >
              Upload
            </a>
          </div>
        </div>
        <hr />
      </header>
      <main>
        <div
          class="videos-list"
        />
        <p
          class="text-center"
        >
          All rights reserved
        </p>
      </main>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="row"
      >
        <div
          class="header-logo"
        >
          <svg>
            logo.svg
          </svg>
          <h1>
            FakeTube
          </h1>
        </div>
        <button
          type="button"
        >
          Login
        </button>
      </div>
      <hr />
      <div
        class="row"
      >
        <div
          class="menu"
        >
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            href="/"
          >
            Home
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            href="/videos"
          >
            Videos
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            href="/channels"
          >
            Channels
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            href="/history"
          >
            History
          </a>
          <a
            class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1iwtne7-MuiTypography-root-MuiLink-root"
            href="/upload"
          >
            Upload
          </a>
        </div>
      </div>
      <hr />
    </header>
    <main>
      <div
        class="videos-list"
      />
      <p
        class="text-center"
      >
        All rights reserved
      </p>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
